pkg/util/hostname/providers.go

```diff
@@ -5,13 +5,33 @@
 
 package hostname
 
-// Provider is a generic function to grab the hostname and return it
-type Provider func(string) (string, error)
+// ProviderMethod is a generic function to grab the hostname and return it
+type ProviderMethod func(string) (string, error)
 
-// ProviderCatalog holds all the various kinds of hostname providers
-var ProviderCatalog = make(map[string]Provider)
+// ProviderPriority allows to specify a lookup priority for a provider
+type ProviderPriority uint
```

Comment:
This is way overkill. As @tmichelet mentioned above we should probably just hardcode providers.